processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -0,0 +1,45 @@
+package io.druid.query.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+public class InDimFilter implements DimFilter
+{
+  private final List<String> values;
+  private final String dimension;
+
+  @JsonCreator
+  public InDimFilter(@JsonProperty("dimension") String dimension, @JsonProperty("values") List<String> values)
+  {
+    this.values = values;
+    this.dimension = dimension;
+  }
+
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  public List<String> getValues()
+  {
+    return values;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+
+    byte[] dimensionBytes = StringUtils.toUtf8(dimension);
+    byte[] valueBytes = (values == null) ? new byte[]{} : StringUtils.toUtf8(values.toString());
```

Comment:
Usually its corner cases like people putting quotes and/or commas in their values.

The "safe" thing to do (and what we do elsewhere) is to add each value individually with an invalid UTF8 character (like `0xFF`) after each string.